lib/irb/statement.rb

```diff
@@ -61,17 +61,16 @@ def is_assignment?
         @is_assignment
       end
 
-      def evaluable_code
-        @code
+      def execute(context, line_no)
+        context.evaluate(@code, line_no)
```

Comment:
IMO, statement objects should only provide information and the evaluation should still happen in the `Irb` objects.

Reply:
Then we need to introduce this kind of branch, because we can't provide evaluable_code for command.
```ruby
case statement
when Statement::Expression
  context.evaluate(statement.code, line_no)
when Statement::Command
  ret = statement.command_class.execute(context, statement.arg)
  context.set_last_value(ret)
end
```
I think `def execute` is better than this case-when.

Reply:
I still prefer having the case statement inside `Irb#eval_input` than making statements prefer the evaluation. Otherwise we're risking having a wrong coupling like RubyLex had, which is hard to eradicate as we both know.

Also, in the future (maybe even in the same release), we'll change commands to NOT have a return value. When that happens, it'd be even less beneficial to encapsulate things in statements.